File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.dart
class Solution {
  int maxDepth(String s) {
    List<int> st = [];
    int maxCounter = 0;

    for (int i = 0; i < s.length; i++) {
      if (s[i] == '(') {
        st.add(1);
      } else if (s[i] == ')') {
        st.removeLast();
      }
      maxCounter = maxCounter > st.length ? maxCounter : st.length;
    }

    return maxCounter;
  }
}

File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let mut st = Vec::new();
        let mut max_counter = 0;

        for c in s.chars() {
            if c == '(' {
                st.push(1);
            } else if c == ')' {
                st.pop();
            }
            max_counter = max_counter.max(st.len() as i32);
        }

        max_counter
    }
}
